solr/core/src/test/org/apache/solr/handler/admin/api/GetSchemaAPITest.java

```diff
@@ -55,6 +59,38 @@ public void testReliesOnIndexSchemaWhenFetchingWholeSchema() {
     assertEquals("flagValue", response.schema.get("flagKey"));
   }
 
+  @Test
+  public void testLooksUpNameFromLatestCoreSchema() throws Exception {
+    when(mockSchema.getSchemaName()).thenReturn("expectedSchemaName");
+
+    final GetSchemaAPI.SchemaNameResponse response = api.getSchemaName();
+
+    assertEquals("expectedSchemaName", response.name);
+    assertNull(response.error);
+  }
+
+  /**
+   * Test the v2 to v1 response mapping for /schema/name
+   *
+   * <p>{@link SchemaHandler} uses the v2 {@link GetSchemaAPI} (and its response class {@link
+   * GetSchemaAPI.SchemaNameResponse}) internally to serve the v1 version of this functionality. So
+   * it's important to make sure that our response stays compatible with SolrJ - both because that's
+   * important in its own right and because that ensures we haven't accidentally changed the v1
+   * response format.
+   */
+  @Test
+  public void testResponseCanBeParsedBySolrJ() {
```

Comment:
[+1] Great idea to test the SolrJ compatibility - I should've been doing this in a lot of other places but hadn't thought of it.

Reply:
It was your idea, I just copied this test from the already existing one :D I can add similar tests to more places if you think it would be useful.

Reply:
Yikes - it worries me that I didn't remember haha.

> I can add similar tests to more places if you think it would be useful.

Absolutely, if you're willing. I think this would be a great type of test to have for other APIs as well!